crypto/rand/rand_lib.c

```diff
@@ -754,7 +766,7 @@ EVP_RAND_CTX *RAND_get0_primary(OSSL_LIB_CTX *ctx)
         return ret;
     }
 
-#ifdef FIPS_MODULE
+#if defined(FIPS_MODULE) && defined(OPENSSL_NO_FIPS_JITTER)
```

Comment:
Indeed, And it will safe on submissions too like mentioned in https://github.com/openssl/openssl/pull/25275

Reply:
Inside the FIPS provider, #25415 replaces the primary DRBG with the health tests.  This means that, inside the FIPS provider, no DRBG is feeding another.

Outside the FIPS provider is pretty much irrelevant from NIST's perspective.  I kept the primary DRBG there because it greatly reduces the amount of entropy drawn from the system.

Reply:
#25415 indeed replaces the primary DRBG with CRNGT via

```c
#ifdef FIPS_MODULE
    ret = rand_new_crngt(ctx, dgbl->seed);
#else
    ...
    ret = rand_new_drbg(ctx, dgbl->seed, PRIMARY_RESEED_INTERVAL,
                        PRIMARY_RESEED_TIME_INTERVAL, 1);
#endif
```

However, `#ifdef FIPS_MODULE` becomes `#if defined(FIPS_MODULE) && defined(OPENSSL_NO_FIPS_JITTER)` in this PR.

In the end, only FIPS provider built with `disable-fips-jitter` flag does this replacement.

Maybe we should keep `#ifdef FIPS_MODULE` here rather than change to `#if defined(FIPS_MODULE) ...`?

Reply:
Good catch.  This isn't what I intended to do here.